osu.Game/Screens/Backgrounds/BackgroundScreenDefault.cs

```diff
@@ -48,9 +59,57 @@ private void display(Background newBackground)
             currentDisplay++;
         }
 
+        private void onUserChanged(ValueChangedEvent<User> user)
+        {
+            if (BackgroundMode.Value == MainMenuBackgroundMode.Default)
```

Comment:
These two if statements can be merged into `BackgroundMode.Value == MainMenuBackgroundMode.Default || user.OldValue?.IsSupporter == user.NewValue?.IsSupporter`

Reply:
Also, you can reverse the condition and call `UpdateBackground` when it is true